syft/internal/unionreader/union_reader_test.go

```diff
@@ -28,3 +30,32 @@ func Test_getUnionReader_notUnionReader(t *testing.T) {
 
 	assert.Equal(t, expectedContents, string(b))
 }
+
+type panickingUnionReader struct{}
+
+func (p2 *panickingUnionReader) ReadAt(p []byte, off int64) (n int, err error) {
+	panic("don't call this in your unit test!")
+}
+
+func (p2 *panickingUnionReader) Seek(offset int64, whence int) (int64, error) {
+	panic("don't call this in your unit test!")
+}
+
+func (p2 *panickingUnionReader) Read(p []byte) (n int, err error) {
+	panic("don't call this in your unit test!")
+}
+
+func (p2 *panickingUnionReader) Close() error {
+	panic("don't call this in your unit test!")
+}
+
+func Test_getUnionReader_fileLocationReadCloser(t *testing.T) {
+	// panickingUnionReader is a UnionReader
+	var _ UnionReader = (*panickingUnionReader)(nil)
+	embedsUnionReader := file.NewLocationReadCloser(file.Location{}, &panickingUnionReader{})
+
+	// embedded union reader is returned without "ReadAll" invocation
+	ur, err := GetUnionReader(embedsUnionReader)
+	require.NoError(t, err)
+	require.NotNil(t, ur)
```

Comment:
I think this should hold a reference to the `panickingUnionReader`, e.g. `expectedUnionReader := &panickingUnionReader{}` which is passed to the `NewLocationReadCloser` call and test that's the thing it gets back from the `GetUnionReader` call, e.g. `require.Equal(t, expectedUnionReader, ur)`, WDYT?